netsupportpolicy.md

```diff
@@ -0,0 +1,16 @@
+.NET support rules:
+1. ODL does not support .NET versions that are out-of-support
+2. .NET support has been tested prior to publishing a version with that .NET support
+3. removing support for a .NET version is a breaking change (and therefore requires a major version increment)
+4. adding support for a .NET version is not a breaking change
+5. ODL major versions are support for at least 1 year after release (TODO this claim was made in the last meeting, but i can't find in the existing doc)
+
+open items:
+1. How do we communicate the lifecycle moves for each ODL major version? (from John)
+    1. My proposal is to let the nuget release notes do this for us.
+2. How do we want to handle testing of .NET release candidates? (from many folks on the team)
+    1. My proposal is to test .NET support after the official .NET release, giving a 3 month runway.
+    2. Mike had feedback to use ODL beta releases, and had this comment: https://teams.microsoft.com/l/message/19:meeting_NWViNmQzN2ItMmQ0Mi00OGFmLWIxMzktZGVhY2EwMWMyMzBm@thread.v2/1720547775436?context=%7B%22contextType%22%3A%22chat%22%7D
+        > There should never be a pre-release version of .NET that does not have OData support (either release or, at minimum, beta). Customers should never be prohibited from moving to any supported or pre-release version of .NET because of lack of OData support.
+4. Mike had this comment in the previous meeting: https://teams.microsoft.com/l/message/19:meeting_NWViNmQzN2ItMmQ0Mi00OGFmLWIxMzktZGVhY2EwMWMyMzBm@thread.v2/1720547233614?context=%7B%22contextType%22%3A%22chat%22%7D
+    > Even if the versioning system does not require that we limit ourselves to only doing breaking change releases during LTS releases, in practice I think we should try to align breaking change releases with the LTS releases.

```

Comment:
this is **also** an internal goal

1. we haven't done many breaking changes
2. there is churn for customers, and many of our customers don't have much code churn on their end and so they are less lilely to desire taking fixes

i need to come with an example of a breaking change release that is outside of the .NET cadence

Reply:
we maybe want to do an ODL LTS so that customers can take new .NET versions without ODL breaking changes?